src/Sentry.Unity/UnityLogExceptionConverter.cs

```diff
@@ -0,0 +1,343 @@
+                    if (match.Groups["file"].Success)
+                    {
+                        file = match.Groups["file"].Value;
+                    }
+
+                    int? line = null;
+                    if (match.Groups["line"].Success)
+                    {
+                        line = Convert.ToInt32(match.Groups["line"].Value);
+                    }
+
+                    frames.Add(new SentryStackFrame
+                    {
+                        Module = module,
+                        Function = function,
+                        FileName = file,
+                        LineNumber = line,
+                        InApp = GuessInApp(module, function, wrapper, file),
+                    });
+                }
+            }
+
+            return IntoSentryException(ule.LogType.ToString(), ule.LogString, frames);
+        }
+
+        private static SentryException IntoSentryException(string type, string value, List<SentryStackFrame> frames)
+        {
+            var stacktrace = new SentryStackTrace();
+            foreach (var frame in Enumerable.Reverse(frames))
+            {
+                stacktrace.Frames.Add(frame);
+            }
+
+            return new SentryException
+            {
+                Stacktrace = stacktrace,
+                Type = type,
+                Value = value,
+                Mechanism = new Mechanism
+                {
+                    Handled = false,
+                    Type = "unity.log"
+                }
+            };
+        }
+
+        private static bool? GuessInApp(string module, string function, string? wrapper, string? file)
+        {
+            if (module == "Unity" || module == "KERNEL32" || module == "ntdll" || module == "USER32" || module == "ole32")
+            {
+                return false;
+            }
+
+            if (function.StartsWith("UnityEngine", StringComparison.Ordinal)
+                || function.StartsWith("System", StringComparison.Ordinal))
+            {
+                return false;
+            }
+
+            if (!string.IsNullOrWhiteSpace(wrapper))
+            {
+                return false;
+            }
+
+            if (file != null && file.EndsWith(".c"))
+            {
+                return false;
+            }
+
+            if (module == "Mono JIT Code" && file != null && file.EndsWith(".cs"))
+            {
+                return true;
+            }
+
+            return null;
+        }
+
+        // https://github.com/getsentry/sentry-unity/issues/103
+        private static string StripZeroes(string filename)
+            => filename.Equals("<00000000000000000000000000000000>", StringComparison.OrdinalIgnoreCase)
+                ? string.Empty
+                : filename;
+
+        // TODO: discuss
+        private static string TryResolveFileNameForMono(string fileName)
+        {
+            try
+            {
+                // throws on Mono for <1231231231> paths
```

Comment:
best we check somehow the pattern instead of relying on the try catch. It can stay just a last resort though